examples/resource-forall.cpp

```diff
@@ -0,0 +1,349 @@
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+
+#if defined(RAJA_ENABLE_OPENMP)
+//----------------------------------------------------------------------------//
+// RAJA::omp_for_parallel_exec policy execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA omp_parallel<seq_exec> vector addition...\n";
+
+  RAJA::forall<RAJA::omp_parallel_exec<RAJA::seq_exec>>(host, RAJA::RangeSegment(0, N), [=] RAJA_DEVICE (int i) { 
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+
+//----------------------------------------------------------------------------//
+// RAJA::omp_for_nowait_exec policy execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA omp_for_nowait vector addition...\n";
+
+  RAJA::forall<RAJA::omp_for_nowait_exec>(host, RAJA::RangeSegment(0, N), [=] (int i) { 
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+
+//----------------------------------------------------------------------------//
+// RAJA::omp_for_exec policy execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA omp_for_exec vector addition...\n";
+
+  RAJA::forall<RAJA::omp_for_exec>(host, RAJA::RangeSegment(0, N), [=] (int i) { 
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+#endif
+
+#if defined(RAJA_ENABLE_TBB)
+//----------------------------------------------------------------------------//
+// RAJA::tbb_for_dynamic policy execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA tbb_for_dynamic vector addition...\n";
+
+  RAJA::forall<RAJA::tbb_for_dynamic>(host, RAJA::RangeSegment(0, N), [=] (int i) { 
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+
+//----------------------------------------------------------------------------//
+// RAJA::tbb_for_static policy execution.... 
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running RAJA tbb_for_static<8> vector addition...\n";
+
+  RAJA::forall<RAJA::tbb_for_static<8>>(host, RAJA::RangeSegment(0, N), [=] (int i) { 
+    c[i] = a[i] + b[i]; 
+  });
+
+  checkResult(c, N);
+#endif
+
+
+
+#if defined(RAJA_ENABLE_CUDA) || defined(RAJA_ENABLE_HIP)
+
+/*
+  GPU_BLOCK_SIZE - specifies the number of threads in a CUDA/HIP thread block
+*/
+const int GPU_BLOCK_SIZE = 256;
+
+//----------------------------------------------------------------------------//
+// RAJA::cuda/hip_exec policy execution.... 
+//----------------------------------------------------------------------------//
+{
+  std::cout << "\n Running RAJA GPU vector addition on 2 seperate streams...\n";
+#if defined(RAJA_ENABLE_CUDA)
+  RAJA::resources::Cuda res_gpu1;
+  RAJA::resources::Cuda res_gpu2;
+  using EXEC_POLICY = RAJA::cuda_exec<GPU_BLOCK_SIZE, true>;
```

Comment:
may be more clear to use cuda_exec_async here.

Reply:
Done.